### PR TITLE
Implement SearchInput UI component

### LINKED ISSUE #6922: Implement SearchInput UI component

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

- [x] move the SearchInput component to the ui package
- [x] add more unit tests on the component

### What issues does this PR fix or reference?

Fixes #6922 

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by deboer-tim — COMMENTED** [review]
Change and new tests look good. Just wondering if this should be in /input though? e.g. it was proposed we put ErrorMessage/WarningMessage/etc in /alert, and this is the same where there will be a group of similar Input-based components.

File: packages/ui/package.json [hunk 1]
```diff
@@ -55,6 +55,10 @@
     "./ErrorMessage": {
       "types": "./dist/alert/ErrorMessage.svelte.d.ts",
       "svelte": "./dist/alert/ErrorMessage.svelte"
+    },
+    "./SearchInput": {
+      "types": "./dist/input/SearchInput.svelte.d.ts",
+      "svelte": "./dist/input/SearchInput.svelte"
```
Comment (axel7083) [inline packages/ui/package.json]: we might want to rename to `inputs` since we already have other items in it, make sense to use plural

**Review by axel7083 — APPROVED** [review]
LGTM, thanks for the change! 🚀

**Review by deboer-tim — APPROVED** [review]
Thanks for the change!

**Pull request merged**